### PR TITLE
Merge loc build changes into xplat

### PR DESCRIPTION
**Pull request by rainersigwald**
I'll annotate the interesting parts.

### CODE REVIEW

File: build/dirs.proj [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="dir.props" />
+
+  <ItemGroup>
+    <Project Include="LocalizationTasks\Microsoft.Build.LocalizationTasks.csproj"
+             Condition="'$(FullFrameworkBuild)' == 'true'" />
```
Comment (rainersigwald) [inline build/dirs.proj]: I added this condition, but is it the right one? Or is the right thing to build this project if we could use it (host == full framework), and always build it for the current host arch, overriding the build flavor?

File: src/MSBuild.sln [hunk 2]
```diff
@@ -58,10 +58,16 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "nuget", "nuget", "{144F834D
...
+<<<<<<< HEAD
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Dependency", "..\Samples\Dependency\Dependency.csproj", "{9EAE36C3-50CD-49A6-9CA2-94649125DCD1}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TaskWithDependency", "..\Samples\TaskWithDependency\TaskWithDependency.csproj", "{09EEB69D-817E-4B33-9F61-24FC9DA5B3D4}"
 EndProject
+||||||| merged common ancestors
+=======
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Build.LocalizationTasks", "..\build\LocalizationTasks\Microsoft.Build.LocalizationTasks.csproj", "{06347A6C-32A4-4218-A930-0A22699C726C}"
+EndProject
+>>>>>>> 941c5df
```
Comment (rainersigwald) [inline src/MSBuild.sln]: whoops!

File: src/Shared/UnitTests/MockLogger.cs [hunk 3]
```diff
@@ -416,7 +416,7 @@ static private ResourceManager EngineResourceManager
             {
                 if (s_engineResourceManager == null)
                 {
-                    s_engineResourceManager = new ResourceManager("Microsoft.Build.Resources.Strings", typeof(ProjectCollection).GetTypeInfo().Assembly);
+                     s_engineResourceManager = new ResourceManager("Microsoft.Build.Strings", typeof(ProjectCollection).GetTypeInfo().Assembly);
```
Comment (rainersigwald) [inline src/Shared/UnitTests/MockLogger.cs]: Pretty sure this is the right change, right? Combining `GetTypeInfo` from xplat with the rename from master?

File: targets/BootStrapFullFrameworkMSBuild.proj [hunk 4]
```diff
@@ -19,7 +19,7 @@
 
     <InstalledNuGetFiles Include="$(MSBuildExtensionsPath)\Microsoft\NuGet\*" />
 
-    <FreshlyBuiltBinaries Include="$(DeploymentDir)*.dll" />
+    <FreshlyBuiltBinaries Include="$(DeploymentDir)**\*.dll" />
```
Comment (rainersigwald) [inline targets/BootStrapFullFrameworkMSBuild.proj]: This conflicted.

**Comment by rainersigwald** [comment]
Well, that's a new failure:

```
MESSAGE:
Microsoft.Build.Shared.InternalErrorException : MSB0001: Internal MSBuild Error: Request 0 was expected to be in state Executing but is not scheduled at all (it may be unscheduled or may be unknown to the system.)
+++++++++++++++++++
STACK TRACE:
at Microsoft.Build.Execution.BuildManager.EndBuild() in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\BackEnd\BuildManager\BuildManager.cs:line 671 at Microsoft.Build.Execution.BuildManager.Build(BuildParameters parameters, BuildRequestData requestData) in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\BackEnd\BuildManager\BuildManager.cs:line 698 at Microsoft.Build.Execution.ProjectInstance.Build(String[] targets, IEnumerable`1 loggers, IEnumerable`1 remoteLoggers, ILoggingService loggingService, Int32 maxNodeCount, IDictionary`2& targetOutputs) in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\Instance\ProjectInstance.cs:line 1839 at Microsoft.Build.Evaluation.Project.Build(String[] targets, IEnumerable`1 loggers, IEnumerable`1 remoteLoggers) in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\Definition\Project.cs:line 1758 at Microsoft.Build.Evaluation.Project.Build(String target, IEnumerable`1 loggers, IEnumerable`1 remoteLoggers) in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\Definition\Project.cs:line 1704 at Microsoft.Build.Evaluation.Project.Build(String target, IEnumerable`1 loggers) in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\Definition\Project.cs:line 1689 at Microsoft.Build.UnitTests.Definition.ItemDefinitionGroup_Tests.MetadataConditionOnItemDefinitionChildBothQualifiedAndUnqualified() in D:\j\workspace\_Windows_NT_C---1e354809\src\XMakeBuildEngine\UnitTests\Definition\ItemDefinitionGroup_Tests.cs:line 973
```

@dotnet-bot test Windows_NT Build for CoreCLR please

**Comment by rainersigwald** [comment]
@cdmihai can you take a look?

File: build/dirs.proj (hunk 1 above)
Comment (cdmihai) [inline build/dirs.proj]: I'd say `fullframework == true && LocalizedBuild == true` should be good enough.

File: src/Shared/UnitTests/MockLogger.cs (hunk 3 above)
Comment (cdmihai) [inline src/Shared/UnitTests/MockLogger.cs]: The new convention is `<AssemblyName>.<NeutralResxName>.resources`: https://github.com/Microsoft/msbuild/blob/master/build/LocalizationTasks/EmitLocalizedResources.cs#L72

For this diff, the resource manager most likely won't be able to find `Microsoft.Build.Strings`.

Comment (cdmihai) [inline src/Shared/UnitTests/MockLogger.cs]: Scratch that, you're right. I had gotten temporarily confused. Even though the .resource file name is `<AssemblyName>.<NeutralResxName>.resources`, the CLR strips away  the `.resources` part of the string.

File: targets/BootStrapFullFrameworkMSBuild.proj (hunk 4 above)
Comment (cdmihai) [inline targets/BootStrapFullFrameworkMSBuild.proj]: That's correct, it now needs the `**` to find the one level deeper satellite assemblies and to reconstruct the recursive dir path.

File: dir.props [hunk 5]
```diff
@@ -332,6 +332,7 @@
   <!-- Change define constants as needed -->
   <PropertyGroup Condition="'$(MsbuildDebugger)' == 'true' AND '$(MonoBuild)' != 'true'">
     <DefineConstants>$(DefineConstants);FEATURE_MSBUILD_DEBUGGER</DefineConstants>
+    <DefineConstants Condition="$(LocalizedBuild) == 'true'">$(DefineConstants);LOCALIZED_BUILD</DefineConstants>
```
Comment (cdmihai) [inline dir.props]: I think the `Localized_Build` define needs to be taken out. During release builds, I doubt that `MSBuildDebugger` is set to true. Or is it? What's that anyway?

Comment (rainersigwald) [inline dir.props]: Yikes, yeah--that's obviously wrong.

It's talking about the half-implemented feature to step through MSBuild builds in the VS debugger, which is off in all of our builds.

**Comment by cdmihai** [comment]
That is indeed a new failure :) Can't see any connection to localized builds. Let's see if it repeats.

---
In reply to: [239257510](https://github.com/Microsoft/msbuild/pull/893#issuecomment-239257510) [](ancestors = 239257510)

**Comment by rainersigwald** [comment]
I accidentally went all the way to master (like #898). Fixed with a force-push.

File: src/Shared/UnitTests/MockLogger.cs (hunk 3 above)
Comment (rainersigwald) [inline src/Shared/UnitTests/MockLogger.cs]: Yeah, this was a real merge-conflict head-scratcher :)

**Comment by cdmihai** [comment]
:shipit:

**Comment by rainersigwald** [comment]
Validated locally that a loc build passed all tests and updated the .xlf files.

**Pull request merged**